src/vm/eetoprofinterfaceimpl.cpp

```diff
@@ -3184,6 +3230,78 @@ HRESULT EEToProfInterfaceImpl::JITCompilationStarted(FunctionID functionId,
     }
 }
 
+HRESULT EEToProfInterfaceImpl::DynamicMethodJITCompilationFinished(FunctionID functionId)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+        CAN_TAKE_LOCK;
+
+        // The JIT / MethodDesc code likely hold locks while this callback is made
+
+        SO_NOT_MAINLINE;
+    }
+    CONTRACTL_END;
+    
+    CLR_TO_PROFILER_ENTRYPOINT((LF_CORPROF, 
+                                LL_INFO1000, 
+                                "**PROF: DynamicMethodJITCompilationFinished 0x%p.\n", 
+                                functionId));
+
+    _ASSERTE(functionId);
+
+    // Should only be called on profilers that support ICorProfilerCallback8
+    if (m_pCallback8 == NULL)
+    {
+        return E_FAIL;
+    }
+
+    {
+        // All callbacks are really NOTHROW, but that's enforced partially by the profiler,
+        // whose try/catch blocks aren't visible to the contract system        
+        PERMANENT_CONTRACT_VIOLATION(ThrowsViolation, ReasonProfilerCallout);
+        return m_pCallback8->DynamicMethodJITCompilationFinished(functionId);
+    }
+}
+
+HRESULT EEToProfInterfaceImpl::DynamicMethodJITCompilationStarted(FunctionID functionId, LPCBYTE ilHeader)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+        CAN_TAKE_LOCK;
+
+        // The JIT / MethodDesc code likely hold locks while this callback is made
+
+        SO_NOT_MAINLINE;
+    }
+    CONTRACTL_END;
+    
+    CLR_TO_PROFILER_ENTRYPOINT((LF_CORPROF, 
+                                LL_INFO1000, 
+                                "**PROF: DynamicMethodJITCompilationStarted 0x%p.\n", 
+                                functionId));
+
+    _ASSERTE(functionId);
+
+    // Should only be called on profilers that support ICorProfilerCallback8
+    if (m_pCallback8 == NULL)
+    {
+        return E_FAIL;
```

Comment:
S_OK seems better (or just make this a void method since the return result is ignored anyways)